src/Playwright/Core/ElementHandle.cs

```diff
@@ -50,96 +50,106 @@ internal ElementHandle(IChannelOwner parent, string guid, ElementHandleInitializ
 
         internal IChannel<ElementHandle> ElementChannel => _channel;
 
-        public async Task<IElementHandle> WaitForSelectorAsync(string selector, WaitForSelectorState? state, float? timeout)
+        public async Task<IElementHandle> WaitForSelectorAsync(string selector, ElementHandleWaitForSelectorOptions options = default)
             => (await _channel.WaitForSelectorAsync(
                 selector: selector,
-                state: state,
-                timeout: timeout).ConfigureAwait(false))?.Object;
+                state: options?.State,
+                timeout: options?.Timeout).ConfigureAwait(false))?.Object;
 
-        public Task WaitForElementStateAsync(ElementState state, float? timeout)
-            => _channel.WaitForElementStateAsync(state, timeout);
+        public Task WaitForElementStateAsync(ElementState state, ElementHandleWaitForElementStateOptions options = default)
+            => _channel.WaitForElementStateAsync(state, timeout: options?.Timeout);
 
-        public Task PressAsync(string key, float? delay, bool? noWaitAfter, float? timeout)
-            => _channel.PressAsync(key, delay, timeout, noWaitAfter);
+        public Task PressAsync(string key, ElementHandlePressOptions options = default)
+            => _channel.PressAsync(
+                key,
+                delay: options?.Delay,
+                timeout: options?.Timeout,
+                noWaitAfter: options?.NoWaitAfter);
 
-        public Task TypeAsync(string text, float? delay, bool? noWaitAfter, float? timeout)
-            => _channel.TypeAsync(text, delay, timeout, noWaitAfter);
+        public Task TypeAsync(string text, ElementHandleTypeOptions options = default)
+            => _channel.TypeAsync(text, delay: options?.Delay, timeout: options?.Timeout, noWaitAfter: options?.NoWaitAfter);
 
-        public async Task<byte[]> ScreenshotAsync(
-            string path,
-            ScreenshotType? type,
-            int? quality,
-            bool? omitBackground,
-            float? timeout)
+        public async Task<byte[]> ScreenshotAsync(ElementHandleScreenshotOptions options = default)
         {
-            if (type == null && !string.IsNullOrEmpty(path))
+            options ??= new();
+            if (options.Type == null && !string.IsNullOrEmpty(options.Path))
             {
-                type = DetermineScreenshotType(path);
+                options.Type = DetermineScreenshotType(options.Path);
             }
 
-            byte[] result = Convert.FromBase64String(await _channel.ScreenshotAsync(path, omitBackground, type, quality, timeout).ConfigureAwait(false));
+            byte[] result = Convert.FromBase64String(await _channel.ScreenshotAsync(
+                options.Path,
```

Comment:
Sometimes you pass them as named, but sometimes as positional args. What should it be? Also, every time I see options passed as is I feel like Channel could just serialize them reflectively.